betterrolls5e/scripts/utils/utils.js

```diff
@@ -494,7 +494,7 @@ export class ItemUtils {
 	static ensureFlags(item) {
 		const flags = this.createFlags(item?.data);
 		if (!flags) return;
-		item.data.update( { flags: { betterRolls5e: flags } } );
+		item.data.flags.betterRolls5e = flags;
```

Comment:
I thought you had to use update, otherwise the document might not get updated properly?

Reply:
And why update when there is a setFlag method? 
https://foundryvtt.com/api/Item.html#setFlag

Reply:
You'll see if you read further up. Tool bonuses will not work if we persist any updates to the item. The reason is because doing so deletes the item's prepared proficiency bonus. Using update() will cause tool rolls to not calculate correctly.

For setFlags specifically though, its because better rolls flags property predates the convention in setFlag(), so the auto used module key name is not 100% the same.

It helps that we don't need to actually set any flags. We're trying to pull flags, and falling back to default data. In fact the correct refactor might be instead to make a extractFlags() method and pass the result around rather than updating the item at all.

Reply:
This explains that! Thanks :)